File: backend/src/main/java/com/legaltrack/service/ChatService.java
package com.legaltrack.service;

import com.legaltrack.dto.chat.ConversationDto;
import com.legaltrack.dto.chat.MessageDto;
import com.legaltrack.dto.chat.SendMessageRequest;

import java.util.List;

public interface ChatService {
    List<ConversationDto> getMyConversations();
    ConversationDto getOrCreateConversation(Long lawyerId, Long caseId);
    List<MessageDto> getConversationMessages(Long conversationId);
    MessageDto sendMessage(Long conversationId, SendMessageRequest request);
    void markConversationAsRead(Long conversationId);
}

File: frontend/src/services/chatService.js
import api from './api';

export const chatService = {
  getMyConversations: async () => {
    const res = await api.get('/chat/conversations');
    return res.data;
  },

  getOrCreateConversation: async (lawyerId, caseId = null) => {
    const res = await api.post(`/chat/conversations?lawyerId=${lawyerId}${caseId ? `&caseId=${caseId}` : ''}`);
    return res.data;
  },

  getMessages: async (conversationId) => {
    const res = await api.get(`/chat/conversations/${conversationId}/messages`);
    return res.data;
  },

  sendMessage: async (conversationId, messageData) => {
    const res = await api.post(`/chat/conversations/${conversationId}/messages`, messageData);
    return res.data;
  },

  markAsRead: async (conversationId) => {
    const res = await api.put(`/chat/conversations/${conversationId}/read`);
    return res.data;
  }
};
